src/wallet/wallet2.cpp

```diff
@@ -3557,10 +3612,156 @@ std::vector<wallet2::pending_tx> wallet2::create_transactions(std::vector<crypto
   }
 }
 
+
+bool wallet2::tx_add_fake_output(std::vector<std::vector<tools::wallet2::get_outs_entry>> &outs, uint64_t global_index, const crypto::public_key& tx_public_key, const rct::key& mask, uint64_t real_index, bool unlocked) const
+{
+  if (!unlocked) // don't add locked outs
+    return false;
+  if (global_index == real_index) // don't re-add real one
+    return false;
+  auto item = std::make_tuple(global_index, tx_public_key, mask);
+  if (std::find(outs.back().begin(), outs.back().end(), item) != outs.back().end()) // don't add duplicates
+    return false;
+  outs.back().push_back(item);
+  return true;
+}
+
+bool wallet2::light_wallet_parse_rct_str(const std::string& rct_string, const crypto::public_key& tx_pub_key, uint64_t internal_output_index, rct::key& decrypted_mask, rct::key& rct_commit, bool decrypt) const
+{
+  // rct string is empty if output is non RCT
+  if (rct_string.empty())
+    return false;
+  // rct_string is a string with length 64+64+64 (<rct commit> + <encrypted mask> + <rct amount>)
+  rct::key encrypted_mask;
+  string_tools::hex_to_pod(rct_string.substr(0,64), rct_commit);
+  string_tools::hex_to_pod(rct_string.substr(64,64), encrypted_mask);
```

Comment:
Should test whether rct_string is the correct size, and for errors (ie, non hex input)